MmSupervisorPkg/Core/aux_config.spam.cfg

```diff
@@ -1 +1,437 @@
+validation.reference = "@LongMode"
+validation.type = "self"
+offset = -6
+size = 4
+
+[[rule]]
+symbol = "gPatchSmmInitStack"
+validation.type = "non zero"
+offset = -8
+size = 8
+
+[[rule]]
+symbol = "@L1"
+validation.type = "self"
+validation.reference = "SmmStartup"
+size = 8
+
+[[rule]]
+symbol = "gPatchRebasedFlagAddr32"
+validation.type = "non zero"
+offset = -4
+size = 4
+
+[[rule]]
+symbol = "gPatchSmmRelocationOriginalAddressPtr32"
+validation.type = "self"
+validation.reference = "mSmmRelocationOriginalAddress"
+offset = -4
+size = 4
+
+# TODO: Really should not do this. We need to update
+# the flow to not patch these values.
+[[rule]]
+symbol = "gPatchSmbase"
+validation.type = "non zero"
+offset = -4
+size = 4
+
+[[rule]]
+symbol = "gPatchSmiStack"
+validation.type = "non zero"
+offset = -4
+size = 4
+
+[[rule]]
+symbol = "gPatchSmiCr3"
+validation.type = "non zero"
+offset = -4
+size = 4
+
+[[rule]]
+symbol = "gPatchMsrIa32MiscEnableSupported"
+validation.type = "non zero"
+offset = -4
+size = 4
+
+[[rule]]
+symbol = "SmiHandlerIdtrAbsAddr"
+validation.type = "self"
+validation.reference = "gSmiHandlerIdtr"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "mPatchCetSupported"
+validation.type = "content"
+validation.content = [0x00]
+offset = -1
+size = 1
+
+[[rule]]
+symbol = "CpuSmmDebugEntryAbsAddr"
+validation.type = "self"
+validation.reference = "CpuSmmDebugEntry"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "SmiRendezvousAbsAddr"
+validation.type = "self"
+validation.reference = "SmiRendezvous"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "CpuSmmDebugExitAbsAddr"
+validation.type = "self"
+validation.reference = "CpuSmmDebugExit"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "mCetSupportedAbsAddr"
+validation.type = "self"
+validation.reference = "mCetSupported"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "mXdSupportedAbsAddr"
+validation.type = "self"
+validation.reference = "mXdSupported"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "gcSmiIdtr"
+size = 6
+validation.type = "non zero"
+
+[[rule]]
+symbol = "__security_cookie"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "MmStartupThisAp"
+validation.type = "self"
+validation.reference = "SmmStartupThisAp"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "CurrentlyExecutingCpu"
+validation.type = "none"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "NumberOfCpus"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "CpuSaveStateSize"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "CpuSaveState"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "MmConfigurationTable"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "gMmCoreMmst"
+field = "NumberOfTableEntries"
+validation.type = "non zero"
+
+[[rule]]
+symbol = "mFirstMmi"
+validation.type = "content"
+validation.content = [0x00]
+
+# TODO: Need to handle this one better
+[[rule]]
+symbol = "mMmCoreMmiHandlers"

```

Comment:
Is there any functionality missing from the tool that is making you specify the size here?

Reply:
Good question, mMmCoreMmiHandlers is set as `MM_CORE_MMI_HANDLERS  mMmCoreMmiHandlers[]` with specified init value. We do not have a way to specify which entry in the list we want to validate, right? i.e. symbol = "mMmCoreMmiHandlers[2]", etc?

Reply:
Nope, the tool does not have that capability at this time!

Reply:
No worries, we will get to this piece later in the process :)